src/parser/scanner_support.c

```diff
@@ -617,12 +617,28 @@ pt_check_hint (const char *text, PT_HINT hint_table[], PT_HINT_ENUM * result_hin
 				  if (arg)
 				    {
 				      temp = strstr (arg_start, ".");
+
+				      /*
+				       * "user_name + dot(.)" is added in front of the table name to distinguish
+				       * the user schema. So we cannot separate index names with dot(.).
+				       * dot(.) does not appear more than 2 times, and the last name is the index name.
+				       */
+				      char *resolved_buf = NULL;
+				      if (temp && strstr (temp + 1, ".") != NULL)
+					{
+					  temp = strstr (temp + 1, ".");
+					}
+
 				      if (temp && temp < &(hint_p[j]) && !IS_HINT_ON_TABLE (hint_table[i].hint))
 					{
+
 					  *temp = '\0';
 					  arg->info.name.resolved = pt_trim_as_identifier (arg_start);
 					  arg->info.name.resolved = pt_makename (arg->info.name.resolved);
-					  *temp++ = '.';
+					  *temp = '.';
+
+					  *temp++;
+
```

Comment:
```suggestion
					  *temp++ = '.';
```

Reply:
Are there any other benefits besides shorter code?
Wouldn't it be more readable to write it separately?

Reply:
If you want to separate assigning value and increment pointer, `*temp++;` seems not the intended code. No need to use dereference (`*`). `temp++;`.

Reply:
I used dereference unnecessarily while separating existing code.
Is it ok if I do it like below without using dereference?

```
*temp = '.';
temp++;
```

Reply:
{*temp = '1'; *temp++;} and {*temp = '2';} are equivalent.
{*temp++ = '1';} and {*temp = '1'; temp++;} is the same.

Reply:
"*temp++;" was the wrong change.
To improve readability, I wanted to separate value assignment and repositioning.